src/components/checkbox/checkbox.scss

```diff
@@ -0,0 +1,401 @@
+/** The color of the checkbox border. */
+$md-checkbox-border-color: if($md-is-dark-theme, rgba(white, 0.7), rgba(black, 0.54)) !default;
+/** The color of the checkbox's checkmark / mixedmark. */
+$md-checkbox-mark-color: map-get($md-background, background);
+/** The color that is used as the checkbox background when it is checked. */
+$md-checkbox-background-color: md-color($md-primary, 500);
+/** The base duration used for the majority of transitions for the checkbox. */
+$md-checkbox-transition-duration: 140ms;
+/** The amount of spacing between the checkbox and its label. */
+$md-checkbox-item-spacing: 4px;
+
+// Manual calculation done on SVG
+$_md-checkbox-mark-path-length: 22.910259;
+$_md-checkbox-indeterminate-checked-easing-function: cubic-bezier(0.14, 0, 0, 1);
+
+@keyframes fade-in-background {
+  0% {
+    opacity: 0;
+  }
+
+  50% {
+    opacity: 1;
+  }
+}
+
+@keyframes fade-out-background {
+  0%, 50% {
+    opacity: 1;
+  }
+
+  100% {
+    opacity: 0;
+  }
+}
+
+@keyframes unchecked-checked-checkmark-path {
+  0%, 50% {
+    stroke-dashoffset: $_md-checkbox-mark-path-length;
+  }
+
+  50% {
+    animation-timing-function: $linear-out-slow-in-timing-function;
+  }
+
+  100% {
+    stroke-dashoffset: 0;
+  }
+}
+
+@keyframes unchecked-indeterminate-mixedmark {
+  0%, 68.2% {
+    transform: scaleX(0);
+  }
+
+  68.2% {
+    animation-timing-function: cubic-bezier(0, 0, 0, 1);
+  }
+
+  100% {
+    transform: scaleX(1);
+  }
+}
+
+@keyframes checked-unchecked-checkmark-path {
+  from {
+    animation-timing-function: $fast-out-linear-in-timing-function;
+    stroke-dashoffset: 0;
+  }
+
+  to {
+    stroke-dashoffset: $_md-checkbox-mark-path-length * -1;
+  }
+}
+
+@keyframes checked-indeterminate-checkmark {
+  from {
+    animation-timing-function: $linear-out-slow-in-timing-function;
+    opacity: 1;
+    transform: rotate(0deg);
+  }
+
+  to {
+    opacity: 0;
+    transform: rotate(45deg);
+  }
+}
+
+@keyframes indeterminate-checked-checkmark {
+  from {
+    animation-timing-function: $_md-checkbox-indeterminate-checked-easing-function;
+    opacity: 0;
+    transform: rotate(45deg);
+  }
+
+  to {
+    opacity: 1;
+    transform: rotate(360deg);
+  }
+}
+
+@keyframes checked-indeterminate-mixedmark {
+  from {
+    animation-timing-function: $linear-out-slow-in-timing-function;
+    opacity: 0;
+    transform: rotate(-45deg);
+  }
+
+  to {
+    opacity: 1;
+    transform: rotate(0deg);
+  }
+}
+
+@keyframes indeterminate-checked-mixedmark {
+  from {
+    animation-timing-function: $_md-checkbox-indeterminate-checked-easing-function;
+    opacity: 1;
+    transform: rotate(0deg);
+  }
+
+  to {
+    opacity: 0;
+    transform: rotate(315deg);
+  }
+}
+
+@keyframes indeterminate-unchecked-mixedmark {
+  0% {
+    animation-timing-function: linear;
+    opacity: 1;
+    transform: scaleX(1);
+  }
+
+  32.8%, 100% {
+    opacity: 0;
+    transform: scaleX(0);
+  }
+}
+
+%md-checkbox-cover-element {
```

Comment:
Could you add a brief comment on the role of each placeholder / class where it's not obvious? E.g., what the "cover-element" or "checkbox-frame" are used for.